src/sundials/sundials_utils.h

```diff
@@ -86,4 +89,56 @@ static inline void sunCompensatedSum(sunrealtype base, sunrealtype inc,
   *sum                      = tmp2;
 }
 
+static inline void sunfprintf_real(FILE* fp, SUNOutputFormat fmt,
+                                   sunbooleantype start, const char* name,
+                                   sunrealtype value)
+{
+  if (fmt == SUN_OUTPUTFORMAT_TABLE)
+  {
+    fprintf(fp, "%-*s = " SUN_REAL_FORMAT_G "\n", SUN_TABLE_WIDTH, name, value);
+  }
+  else
+  {
+    if (!start) { fprintf(fp, ","); }
+    fprintf(fp, "%s," SUN_REAL_FORMAT_E, name, value);
+  }
+}
+
+static inline void sunfprintf_long(FILE* fp, SUNOutputFormat fmt,
+                                   sunbooleantype start, const char* name,
+                                   long value)
+{
+  if (fmt == SUN_OUTPUTFORMAT_TABLE)
+  {
+    fprintf(fp, "%-*s = %ld\n", SUN_TABLE_WIDTH, name, value);
+  }
+  else
+  {
+    if (!start) { fprintf(fp, ","); }
+    fprintf(fp, "%s,%ld", name, value);
+  }
+}
+
+static inline void sunfprintf_long_array(FILE* fp, SUNOutputFormat fmt,
+                                         sunbooleantype start, const char* name,
+                                         long* value, size_t count)
+{
+  if (count < 1) { return; }
+
+  if (fmt == SUN_OUTPUTFORMAT_TABLE)
+  {
+    fprintf(fp, "%-*s = %ld\n", SUN_TABLE_WIDTH, name, value[0]);
```

Comment:
I'd prefer to keep the style where all the values are on the same line
```suggestion
    fprintf(fp, "%-*s = %ld", SUN_TABLE_WIDTH, name, value[0]);
```